dht/range_streamer.cc

```diff
@@ -290,20 +290,19 @@ future<> range_streamer::stream_async() {
                 try {
                     for (auto it = range_vec.begin(); it < range_vec.end();) {
                         ranges_to_stream.push_back(*it);
-                        it = range_vec.erase(it);
+                        ++it;
                         if (ranges_to_stream.size() < nr_ranges_per_stream_plan) {
                             continue;
                         } else {
                             do_streaming();
+                            it = range_vec.erase(range_vec.begin(), it);
```

Comment:
`do_streaming()` is calling `nr_ranges_to_stream()`, which is looking at these `range_vec`s, so it assumes that the `range_vec`s are modified before `do_streaming()` is called.

Reply:
Ouch.  Maybe we need to rely on the local `nr_ranges_remaining` variable instead.
Will look into that...

Reply:
added a `_nr_ranges_remaining` member that is incrementally updated.

Reply:
Could you include that member in this first commit?